feg/gateway/services/n7_n40_proxy/servicers/create_session_test.go

```diff
@@ -0,0 +1,336 @@
+	require.Error(t, err)
+	mockN7.AssertExpectations(t)
+	assert.Nil(t, response)
+}
+
+func TestDisableN7Response(t *testing.T) {
+	srv, mockDb, mockN7 := newCentralSessionControllerForTest(t, true)
+
+	mockDb.On("GetOmnipresentRules").Return(
+		[]string{"omni_rule_1", "omni_rule_2"}, []string{"base_10"}).Once()
+	mockDb.On("GetRuleIDsForBaseNames", []string{"base_10"}).Return([]string{"base_rule_1", "base_rule_2"})
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.NoError(t, err)
+	mockDb.AssertExpectations(t)
+	mockN7.AssertNotCalled(t, "PostSmPoliciesWithResponse")
+
+	assert.Equal(t, genSessionID(IMSI1), response.SessionId)
+	allRuleIds := []string{}
+	for _, staticRule := range response.StaticRules {
+		allRuleIds = append(allRuleIds, staticRule.RuleId)
+	}
+	assert.ElementsMatch(t, []string{"omni_rule_1", "omni_rule_2", "base_rule_1", "base_rule_2"}, allRuleIds)
+}
+
+func newCentralSessionControllerForTest(t *testing.T, disableN7 bool) (*servicers.CentralSessionController, *mockPolicyDB.PolicyDBClient, *mockN7.ClientWithResponsesInterface) {
+	testN7Conf := getTestN7Config(t)
+	testN7Conf.DisableN7 = disableN7
+	mockPolicyDBClient := &mockPolicyDB.PolicyDBClient{}
+	mockPolicyClient := &mockN7.ClientWithResponsesInterface{}
+
+	srv, err := servicers.NewCentralSessionController(testN7Conf, mockPolicyDBClient, mockPolicyClient)
+	require.NoError(t, err)
+	return srv, mockPolicyDBClient, mockPolicyClient
+}
+
+func getTestN7Config(t *testing.T) *n7.N7Config {
+	apiRoot, err := url.ParseRequestURI(URL1)
+	require.NoError(t, err)
+	return &n7.N7Config{
+		DisableN7: false,
+		Server: n7.PCFConfig{
+			ApiRoot:      *apiRoot,
+			TokenUrl:     TOKEN_URL,
+			ClientId:     CLIENT_ID,
+			ClientSecret: CLIENT_SECRET,
+		},
+		Client: n7.N7ClientConfig{
+			LocalAddr:     LOCAL_ADDR,
+			NotifyApiRoot: NOTIFY_API_ROOT,
+		},
+	}
+}
+
+func newCreateSessionRequest(imsi string) *protos.CreateSessionRequest {
+	return &protos.CreateSessionRequest{
+		CommonContext: &protos.CommonSessionContext{
+			Sid:     &protos.SubscriberID{Id: imsi},
+			RatType: protos.RATType_TGPP_NR,
+			UeIpv4:  UE_IPV4,
+			Apn:     APN1,
+		},
+		RatSpecificContext: &protos.RatSpecificContext{
+			Context: &protos.RatSpecificContext_M5GsmSessionContext{
+				M5GsmSessionContext: &protos.M5GSMSessionContext{
+					PduSessionId:   uint32(PDU_SESSION_ID),
+					Gpsi:           GPSI1,
+					PduSessionType: protos.PduSessionType_IPV4,
+				},
+			},
+		},
+		SessionId:      genSessionID(IMSI1),
+		AccessTimezone: &protos.Timezone{OffsetMinutes: 330},
+	}
+}
+
+func genSessionID(imsi string) string {
+	return fmt.Sprintf("%s-1234", imsi)
+}
+
+func matchSmPolicyContextData(imsi string) interface{} {
+	return func(reqBody n7_sbi.PostSmPoliciesJSONRequestBody) bool {
+		return reqBody.Supi == sbi.Supi(imsi) &&
+			reqBody.NotificationUri == n7.GenNotifyUrl(NOTIFY_API_ROOT, genSessionID("IMSI"+imsi))
+	}
+}
```

Comment:
Maybe this is checked somewhere else, but should't you have a matcher to see if all the parameters you requested did make it to the request you are sending?

Reply:
This is done.